RealestateApi/Repository/ReferenceDataRepository.cs

```diff
@@ -213,5 +213,93 @@ public async Task<ReferenceDataResponseDto> DeleteAmenityReferenceDataAsync(stri
 
             return response;
         }
+
+        public async Task<ReferenceDataModel> GetTypologyReferenceDataAsync(string refDataType, int refDataId)
+        {
+            using var conn = await _dataSource.OpenConnectionAsync();
+
+            using var typologyQuery = new NpgsqlCommand("SELECT * FROM typology WHERE id = @RefDataId;", conn);
+            typologyQuery.Parameters.AddWithValue("@RefDataId", refDataId);
+            using var typologyReader = await typologyQuery.ExecuteReaderAsync();
+
+            while (typologyReader.Read())
+            {
+                ReferenceDataModel response = new()
+                {
+                    Id = (int)typologyReader["id"],
+                    Description = (string)typologyReader["description"],
+                };
+
+                return response;
+            }
+
+            return null;
```

Comment:
You should not return null, you should return response.

By returning response inside a while loop, you will return as soon as the first one is read.